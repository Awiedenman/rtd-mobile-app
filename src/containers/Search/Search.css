* {
  box-sizing: border-box;
}
.search_container {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 95%;
  margin-top: 10px;
  background: white;
  box-shadow: 1px 1px 2px 2px rgba(0, 0, 0, 0.5);
}

.search-title {
  margin: 20px;
}

.form {
  width: 100%;
}

.location-search-input {
  height: 40px;
  width: 90%;
  border: 1px solid gray;
  font-size: 15px;
  padding-left: 5px;
}

.autocomplete-dropdown-container {
  position: absolute;
  top: 40px;
  left: 20px;
  width: 89.75%;
  background: #FFF;
  z-index: 20;
  border: 1px solid #808080;
  border-top: none;
}

#end-address {
  border-top: none;
  z-index: 19;
}

.time-select-container {
  display: flex;
  align-items: center;
  margin: 20px 0 20px 10%;
}

.radio-container {
  display: flex;
}

.radio-label {
  font-size: 15px;
}

#at {
  margin: 0 10px 0 5px;
  font-size: 15px;
}

.time-select {
  margin: 5px;
  width: 20px;
  border: 1px solid gray;
}

.search-button {
  color: white;
  font-size: 15px;
  height: 33px;
  width: 100%;
  border: none;
  background: #c02d34;
  border: none;
}

div .location-input {
  /* width: 150%; */
}

