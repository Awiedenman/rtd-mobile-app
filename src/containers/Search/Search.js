import React, { Component } from 'react';
import { fetchScheduleThunk } from '../../thunks/fetchScheduleThunk';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './Search.css';


export class Search extends Component {
  constructor(props){
    super(props);
    this.state = {
      startPoint: '',
      destination: ''
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const url = 'somestring';
    this.props.storeRouteSchedules(url);
  }

  render(){
    return (
      <div className="search_container">
        <form onSubmit={ this.handleSubmit }>
          <input 
            className="starting_location"
            onChange={ this.handleChange }
            type="text" 
            name="startPoint"
            value={ this.state.startPoint } 
            placeholder="Starting Location" 
          />
          <input 
            className = "destination"
            onChange={ this.handleChange }
            type="text" 
            name='destination' 
            value={ this.state.destination } 
            placeholder="Destination" 
          />
          <button 
            className="search_button"
            type='submit'>Search
          </button>
        </form>   
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  storeRouteSchedules: (url) => dispatch(fetchScheduleThunk(url))
});

export default connect(null, mapDispatchToProps)(Search);

Search.propTypes = {
  storeRouteSchedules: PropTypes.func
};