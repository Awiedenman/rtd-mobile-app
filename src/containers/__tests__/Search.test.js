import React from 'react';
import { shallow } from 'enzyme';
import { Search } from '../Search/Search';
// import { history } from 'react-router-dom'

describe('Search', () => {
  let wrapper;
  let mockStoreUserSearch;
  let mockPostItineraryThunk;
  let mockStartAddress;
  let mockEndAddress;
  let mockUid;
  let mockIsLoading;
  let mockHistory;
  let mockTimeData;

  beforeEach(() => {
    mockTimeData = {
      hours: 1,
      minutes: 30,
      departing: true,
      am: true
    }
    mockStoreUserSearch = jest.fn()
    mockPostItineraryThunk = jest.fn()
    mockStartAddress = 'Union Station, Denver, CO';
    mockEndAddress = 'Ruby Hill Park, Denver, CO'
    mockHistory = jest.fn();
    mockUid= '123456abcdefg'
    mockIsLoading=false 
    wrapper = shallow(
      <Search
        postItineraryThunk = {mockPostItineraryThunk}
        storeUserSearch = {mockStoreUserSearch}
        startAddress = {mockStartAddress}
        endAddress = {mockEndAddress}
        uid = {mockUid}
        isLoading = {mockIsLoading}
        am={false}
        departing={false}
        hours={10}
        minutes={30}
      />
    )  
  })

  // it('should match snapshot test', () => {
  //   expect(wrapper).toMatchSnapshot();
  // })

  // it("should match snapshot test if this.state.am is ", () => {
    
  //   wrapper.setState({am: true});
 
  //   expect(wrapper).toMatchSnapshot();
  // });

  describe('handleChange', () => {
    it('should change the value of this.state.departing it ', () => {

      const event = {
        target: {
          name: 'departing',
          value: false
        }
      } 
      const expected = false;

      wrapper.instance().handleChange(event);

      expect(wrapper.state('departing')).toEqual(expected)
    })
  })

  // describe('getTime', () => {
  //   it('should return the correct time', () => {
  //     const hours = 14;

  //     wrapper.instance().getTime()

  //     expect(wrapper.hours).toEqual(2)
  //   })
  // })
  
  describe('handleClick', () => {
    test.skip('should change the state of am from true to false on click', () => {
      const expected = false;

      wrapper.instance().handleClick();

      expect(wrapper.state('am')).toEqual(expected)
    })
  })
  
  describe.skip('handleSubmit', () => {

    it('should call makeOptions', () => {
      const event = {
        preventDefault() {}
      }
      const makeOptions = jest.fn();
      
      wrapper.instance().handleSubmit(event);

      expect(makeOptions).toHaveBeenCalledWith(mockTimeData)
    })
    
  })
  
  describe.skip('makeOptions', () => {
    test('should call itineraryUrl with coorect params', () => {
      const uid = mockUid;
      const itineraryUrl = jest.fn;

      wrapper.instance().makeOptions(uid)

      expect(itineraryUrl).toHaveBeenCalledWith(uid);
    })
    
    it('should call postItineraryThunk', () => {
      const mockPostItineraryThunk = jest.fn();
      const mockTimeData = {
        hours: 1,
        minutes: 30,
        departing: true,
        am: true
      }

      wrapper.instance().makeOptions(mockTimeData)

      expect(mockPostItineraryThunk).toHaveBeenCalled;
    })
  })
})
 

