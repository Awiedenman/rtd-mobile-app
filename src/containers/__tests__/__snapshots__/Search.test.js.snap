// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    am={false}
    departing={false}
    endAddress="Ruby Hill Park, Denver, CO"
    isLoading={false}
    postItineraryThunk={[Function]}
    startAddress="Union Station, Denver, CO"
    storeUserSearch={[Function]}
    uid="123456abcdefg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
>
          Search for a connection:
</h2>,
        <form
          className="form"
          onSubmit={[Function]}
>
          <Connect(StartAddressInput) />
          <Connect(EndAddressInput) />
          <Connect(Time) />
          <button
                    className="button"
                    type="submit"
          >
                    Search
          </button>
</form>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for a connection:",
          "className": "title",
        },
        "ref": null,
        "rendered": "Search for a connection:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(StartAddressInput) />,
            <Connect(EndAddressInput) />,
            <Connect(Time) />,
            <button
              className="button"
              type="submit"
>
              Search
</button>,
          ],
          "className": "form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "button",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
>
            Search for a connection:
</h2>,
          <form
            className="form"
            onSubmit={[Function]}
>
            <Connect(StartAddressInput) />
            <Connect(EndAddressInput) />
            <Connect(Time) />
            <button
                        className="button"
                        type="submit"
            >
                        Search
            </button>
</form>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for a connection:",
            "className": "title",
          },
          "ref": null,
          "rendered": "Search for a connection:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(StartAddressInput) />,
              <Connect(EndAddressInput) />,
              <Connect(Time) />,
              <button
                className="button"
                type="submit"
>
                Search
</button>,
            ],
            "className": "form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "className": "button",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Search should match snapshot test if this.state.am is  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    am={false}
    departing={false}
    endAddress="Ruby Hill Park, Denver, CO"
    isLoading={false}
    postItineraryThunk={[Function]}
    startAddress="Union Station, Denver, CO"
    storeUserSearch={[Function]}
    uid="123456abcdefg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
>
          Search for a connection:
</h2>,
        <form
          className="form"
          onSubmit={[Function]}
>
          <Connect(StartAddressInput) />
          <Connect(EndAddressInput) />
          <Connect(Time) />
          <button
                    className="button"
                    type="submit"
          >
                    Search
          </button>
</form>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for a connection:",
          "className": "title",
        },
        "ref": null,
        "rendered": "Search for a connection:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(StartAddressInput) />,
            <Connect(EndAddressInput) />,
            <Connect(Time) />,
            <button
              className="button"
              type="submit"
>
              Search
</button>,
          ],
          "className": "form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "button",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
>
            Search for a connection:
</h2>,
          <form
            className="form"
            onSubmit={[Function]}
>
            <Connect(StartAddressInput) />
            <Connect(EndAddressInput) />
            <Connect(Time) />
            <button
                        className="button"
                        type="submit"
            >
                        Search
            </button>
</form>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for a connection:",
            "className": "title",
          },
          "ref": null,
          "rendered": "Search for a connection:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(StartAddressInput) />,
              <Connect(EndAddressInput) />,
              <Connect(Time) />,
              <button
                className="button"
                type="submit"
>
                Search
</button>,
            ],
            "className": "form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "className": "button",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
